resources/lib/syncEpisodes.py

```diff
@@ -578,10 +575,10 @@ def __compareShows(self, shows_col1, shows_col2, rating=False, restrict=False):
 
                 if show_col2:
                     show = {'title': show_col1['title'], 'ids': {}, 'year': show_col1['year']}
-                    if 'tvdb' in show_col1['ids']:
-                        show['ids'] = {'tvdb': show_col1['ids']['tvdb']}
-                    if 'imdb' in show_col2 and show_col2['imdb']:
-                        show['ids']['imdb'] = show_col2['imdb']
+                    if show_col1['ids']:
+                        show['ids'].update(show_col1['ids'])
+                    if show_col2['ids']:
+                        show['ids'].update(show_col2['ids'])
```

Comment:
Scrap that then. I guess that the imdbnumber field still gives you the imdb id only, if any. May I ask then why uniqueid is not used for movies and tvshows but only for episodes?

Anyhow, if the match is by id then both col1 and col2 ids should be correct. If matching by title, then there is no coincidental ids between col1 and col2 which would not matter because only those ids coming from trakt would be used whatever is the col they come from. Which might be wrong but that comes from matching by tittle.

The idea of refraining from tittle matching when there is any usable id locally still holds as a suggestion.